### ISSUE TITLE
Put-signal Warning

### ISSUE THREAD

**Issue by jdinan** [Sandia-OpenSHMEM/SOS#829]
I haven't been able to reproduce this warning locally, but I'm seeing the following in the CMA row of Travis.  Travis compiler is almost old enough to vote, so this may be a red herring:
```
[0K[32;1mThe command "../configure --prefix=$TRAVIS_INSTALL/sandia-shmem-cma --with-cma $SOS_GLOBAL_BUILD_OPTS $SOS_BUILD_OPTS" exited with 0.[0m

travis_time:start:04948b36
[0K$ make $TRAVIS_PAR_MAKE
Making all in bindings
make[1]: Entering directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/bindings'
  GEN      shmem_bind_c11_extras.m4
  CXX      generate_bind_cxx_extras.o
  CXXLD    generate_bind_cxx_extras
  GEN      shmem_bind_cxx_extras.m4
make  all-am
make[2]: Entering directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/bindings'
make[2]: Nothing to be done for `all-am'.
make[2]: Leaving directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/bindings'
make[1]: Leaving directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/bindings'
Making all in man
make[1]: Entering directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/man'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/man'
Making all in mpp
make[1]: Entering directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/mpp'
  GEN      shmem.h
  GEN      shmemx.h
make  all-am
make[2]: Entering directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/mpp'
make[2]: Nothing to be done for `all-am'.
make[2]: Leaving directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/mpp'
make[1]: Leaving directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/mpp'
Making all in pmi-simple
make[1]: Entering directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/pmi-simple'
  CC       simple_pmi.lo
  CC       simple_pmiutil.lo
  CCLD     libpmi_simple.la
make[1]: Leaving directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/pmi-simple'
Making all in src
make[1]: Entering directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/src'
  GEN      atomic_c.c
  GEN      atomic_nbi_c.c
  GEN      collectives_c.c
  GEN      data_c.c
  GEN      synchronization_c.c
  GEN      collectives_f.c
  GEN      data_f.c
  GEN      build_info.h
make  all-am
make[2]: Entering directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/src'
  CC       shmem_free_list.lo
  CC       runtime_util.lo
  CC       malloc.lo
  CC       init.lo
  CC       collectives.lo
  CC       init_c.lo
  CC       query_c.lo
  CC       accessibility_c.lo
In file included from ../../src/shmem_accessibility.h:19:0,
                 from ../../src/accessibility_c.c:24:
../../src/shmem_comm.h: In function â€˜shmem_internal_put_signal_nbiâ€™:
../../src/shmem_comm.h:115:13: warning: implicit declaration of function â€˜shmem_internal_membar_storeâ€™ [-Wimplicit-function-declaration]
             shmem_internal_membar_store(); /* Memory fence to ensure target PE observes stores in the correct order */
             ^
  CC       symmetric_heap_c.lo
In file included from ../../src/symmetric_heap_c.c:34:0:
../../src/shmem_comm.h: In function â€˜shmem_internal_put_signal_nbiâ€™:
../../src/shmem_comm.h:115:13: warning: implicit declaration of function â€˜shmem_internal_membar_storeâ€™ [-Wimplicit-function-declaration]
             shmem_internal_membar_store(); /* Memory fence to ensure target PE observes stores in the correct order */
             ^
In file included from ../../src/shmem_synchronization.h:19:0,
                 from ../../src/shmem_collectives.h:19,
                 from ../../src/symmetric_heap_c.c:35:
../../src/shmem_atomic.h: At top level:
../../src/shmem_atomic.h:118:1: warning: conflicting types for â€˜shmem_internal_membar_storeâ€™ [enabled by default]
 shmem_internal_membar_store(void) {
 ^
../../src/shmem_atomic.h:118:1: error: static declaration of â€˜shmem_internal_membar_storeâ€™ follows non-static declaration
In file included from ../../src/symmetric_heap_c.c:34:0:
../../src/shmem_comm.h:115:13: note: previous implicit declaration of â€˜shmem_internal_membar_storeâ€™ was here
             shmem_internal_membar_store(); /* Memory fence to ensure target PE observes stores in the correct order */
             ^
  CC       remote_pointer_c.lo
make[2]: *** [symmetric_heap_c.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from ../../src/shmem_remote_pointer.h:19:0,
                 from ../../src/remote_pointer_c.c:24:
../../src/shmem_comm.h: In function â€˜shmem_internal_put_signal_nbiâ€™:
../../src/shmem_comm.h:115:13: warning: implicit declaration of function â€˜shmem_internal_membar_storeâ€™ [-Wimplicit-function-declaration]
             shmem_internal_membar_store(); /* Memory fence to ensure target PE observes stores in the correct order */
             ^
make[2]: Leaving directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/src'
```

**Comment by wrrobin** [comment]
@jdinan Thanks. I am not sure why it did not occur in the Travis tests (https://travis-ci.org/Sandia-OpenSHMEM/SOS/builds/475393067?utm_source=github_status&utm_medium=notification) when this code is merged. The one you reported failed because of the error?

**Comment by jdinan** [comment]
This is just a warning.  It didn't break the build (write error did that).

**Comment by jdinan** [comment]
Closed by #828
